linux/LibSensors.c

```diff
@@ -229,22 +229,20 @@ void LibSensors_getCPUTemperatures(CPUData* cpus, unsigned int existingCPUs, uns
    }
 
    /* No package temperature - set to max core temperature */
-   if (isnan(data[0]) && coreTempCount != 0) {
-      double maxTemp = NAN;
+   if (coreTempCount > 0 && isNaN(data[0])) {
+      double maxTemp = -HUGE_VAL;
```

Comment:
Ensure `data[0]` is initialized as `NAN` to ensure defined value.

Reply:
I think that is already done in line 147:
```c
   for (size_t i = 0; i < existingCPUs + 1; i++)
      data[i] = NAN;
```

Reply:
k. ACK. Didn't catch that earlier.

Somehow I prefer the original version of that part of the code, as it kept the general pattern closer together. That's not that this version is bad, but a bit more confusing than need be IMHO.